libs/cardano-ledger-api/src/Cardano/Ledger/Api/State/Query.hs

```diff
@@ -185,6 +190,41 @@ queryDRepStakeDistr nes creds
   where
     distr = psDRepDistr . fst $ finishedPulserState nes
 
+-- | Query the stake distribution of the registered DReps. This does not
+-- include the @AlwaysAbstain@ and @NoConfidence@ DReps.
+queryRegisteredDRepStakeDistr ::
+  ConwayEraGov era =>
+  NewEpochState era ->
+  -- | Specify DRep Ids whose stake distribution should be returned. When this set is
+  -- empty, distributions for all of the registered DReps will be returned.
+  Set (Credential 'DRepRole (EraCrypto era)) ->
+  Map (Credential 'DRepRole (EraCrypto era)) Coin
+queryRegisteredDRepStakeDistr nes creds
+  | null creds = distr
+  | otherwise = distr `Map.restrictKeys` creds
+  where
+    distr = Map.map fromCompact $ Map.foldlWithKey' computeDistr mempty registeredDReps
+    registeredDReps = nes ^. nesEsL . esLStateL . lsCertStateL . certVStateL . vsDRepsL
+    computeDistr distrAcc stakeCred (DRepState {..}) =
+      Map.insert
+        stakeCred
+        (addCompact (fromJust $ toCompact drepDeposit) $ totalDelegations drepDelegs)
+        distrAcc
```

Comment:
DRep deposit does not participate in the stake distribution fro voting, it is only the proposal deposit that does. The reason for this even being a thing is because Proposal deposits are huge, while DRep deposit not so much.
```suggestion
      Map.insert dRepCred (totalDelegations drepDelegs) distrAcc
```